packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/task/s3_upload_task.dart

```diff
@@ -574,6 +594,35 @@ class S3UploadTask {
     }
   }
 
+  Future<void> _handleUploadPart({
+    required int partNumber,
+    required Future<_CompletedSubtask> uploadPartRequest,
+  }) {
+    return uploadPartRequest
+        .then(_subtasksStreamController.add)
+        .catchError((Object error) {
+      if (error is CancellationException) {
+        _logger
+            .debug('Part $partNumber upload HTTP operation has been canceled.');
+        return;
+      }
+
+      _terminateMultipartUploadOnError(error, isCancel: false);
+    });
+  }
+
+  Future<void> _cancelOngoingUploadPartOperations({
+    bool cancelingOnPause = false,
+  }) async {
+    for (final operation in _ongoingUploadPartHttpOperations) {
+      await operation.smithyOperation.cancel();
+      if (!cancelingOnPause) {
+        _ongoingSubtasks
+            .removeWhere((task) => task.partNumber == operation.partNumber);
+      }
+    }
+  }
```

Comment:
This should probably be synchronous. `.cancel` does not need to be awaited for Smithy/HTTP ops

```suggestion
  void _cancelOngoingUploadPartOperations({
    bool cancelingOnPause = false,
  }) {
    for (final operation in _ongoingUploadPartHttpOperations) {
      unawaited(operation.smithyOperation.cancel());
      if (!cancelingOnPause) {
        _ongoingSubtasks
            .removeWhere((task) => task.partNumber == operation.partNumber);
      }
    }
  }
```